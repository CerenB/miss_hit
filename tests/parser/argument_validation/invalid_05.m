% From https://www.mathworks.com/help/matlab/matlab_prog/function-argument-validation-1.html

% However, you cannot refer to input variables not yet declared in an
% arguments block. For example, using this declaration for argument a
% in the previous function is not valid because b and c have not been
% declared yet.

function c = invalid_05(a,b,c)
    arguments
        a uint32 = b * c
        b uint32
        c uint32
    end

    % Function code
end
