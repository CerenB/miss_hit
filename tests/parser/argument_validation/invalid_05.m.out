----------------------------------------------------------------------
--  Parse tree for invalid_05.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <invalid_05>
      Inputs: Identifier <a>
      Inputs: Identifier <b>
      Inputs: Identifier <c>
      Outputs: Identifier <c>
    Validation: Arguments Special_Block
      Items: Class_Property
        Name: Identifier <a>
        Class: Identifier <uint32>
        Default: Binary_Operation *
          LHS: Identifier <b>
          RHS: Identifier <c>
      Items: Class_Property
        Name: Identifier <b>
        Class: Identifier <uint32>
      Items: Class_Property
        Name: Identifier <c>
        Class: Identifier <uint32>
    Body: Sequence_Of_Statements
----------------------------------------------------------------------
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
